src/web/wms/src/main/java/org/geoserver/wms/web/data/ExternalGraphicPanel.java

```diff
@@ -292,6 +264,56 @@ protected String baseURL(Form form) {
         return baseUrl;
     }
 
+    /**
+     * Validates the external graphic and returns a connection to the graphic.
+     * If validation fails, error messages will be added to the passed form
+     * 
+     * @param target
+     * @param form
+     * @return URLConnection to the External Graphic file
+     */
+    protected URLConnection getExternalGraphic(AjaxRequestTarget target, Form<?> form) {
+        onlineResource.processInput();
+        if (onlineResource.getModelObject() != null) {
+            URL url = null;
+            try {
+                String baseUrl = baseURL(form);
+                String external = onlineResource.getModelObject().toString();
+                
+                URI uri = new URI( external );
+                if( uri.isAbsolute() ){
+                    url = uri.toURL();
+                    if( !external.startsWith(baseUrl)){
+                        form.warn( "Recommend use of styles directory at "+baseUrl);
+                    }
+                }
+                else {
+                    url = new URL( baseUrl + "styles/"+external );
```

Comment:
This logic changes a bit due to https://github.com/geoserver/geoserver/pull/1749